tests/test_train.py

```diff
@@ -358,3 +359,227 @@ def test_train_nlu_autoconfig(
     mocked_get_configuration.assert_called_once()
     _, args, _ = mocked_get_configuration.mock_calls[0]
     assert args[1] == autoconfig.TrainingType.NLU
+
+
+def mock_async(monkeypatch, target, name):
+    mock = Mock()
+
+    async def mock_async_func(*args: Any, **kwargs: Any) -> None:
+        mock(*args, **kwargs)
+
+    monkeypatch.setattr(target, name, mock_async_func)
+    return mock
+
+
+def mock_core_training(monkeypatch):
+    return mock_async(monkeypatch, rasa.core, rasa.core.train.__name__)
+
+
+def mock_nlu_training(monkeypatch):
+    return mock_async(monkeypatch, rasa.nlu, rasa.nlu.train.__name__)
+
+
+def new_model_path_in_same_dir(old_model_path) -> str:
+    return str(Path(old_model_path).parent / (secrets.token_hex(8) + ".tar.gz"))
+
+
+class TestE2e:
```

Comment:
Maybe makes sense to extract `e2e` to a separate file?

Reply:
It means losing the module name to test file name mapping, but if you think this is too big/different I can move it.

Reply:
I don't feel strongly either way.